pandas/tests/plotting/test_hist_method.py

```diff
@@ -484,3 +506,30 @@ def test_axis_share_xy(self):
 
         assert ax1._shared_y_axes.joined(ax1, ax2)
         assert ax2._shared_y_axes.joined(ax1, ax2)
+
+    @pytest.mark.slow
+    @pytest.mark.parametrize("by", [None, "c"])
+    @pytest.mark.parametrize("column", [None, "b"])
+    @pytest.mark.parametrize("label", [None, "d"])
+    def test_hist_with_legend(self, by, column, label):
+        expected_axes_num = 1 if by is None and column is not None else 2
+        expected_layout = (1, expected_axes_num)
+        expected_labels = label or column or ["a", "b"]
+        if by is not None:
+            expected_labels = [expected_labels] * 2
```

Comment:
similar comments like above, you need parametrize them and see if `slow` mark is needed.

Reply:
also needs to add an issue number

Reply:
also could you rename this function, it has the same name as your the other function above, but they do slightly different thing, right?

also maybe you should combine those two functions probably, their main differences are having `column` fixture or not, which can be added as part of parametrization so that only probably only one function is enough?

Reply:
Regarding the last comment: the difference is that one tests series and other tests frames. They are supposed to be members of TestSeriesPlots and TestDataFramePlots, but I messed up somewhere and attached them to the wrong classes; that will be fixed.

Is being members of TestSeriesPlots and TestDataFramePlots not sufficient to differentiate? Happy to add "df" to the latter name if it is still desired; I personally don't think it's necessary.

But for the sake of consistency with the layout of the tests there, it seems to me like they should be kept separate.

Reply:
ahh, okay, if they are put in different classes, then it's certainly fine!